sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -114,27 +138,48 @@ def fit(self, X, y=None):
         -------
         self
         """
-        feature_names = []
-        vocab = {}
+        return self._partial_fit(X)
+
+    def partial_fit(self, X, y=None):
+        return self._partial_fit(X).self.transform(X)
+
+    def _partial_fit(self, X, y=None):
+        if getattr(self, "feature_names_", None) is None:
+            self.feature_names_ = []
 
+        if getattr(self, "vocabulary_", None) is None:
+            self.vocabulary_ = {}
+            
+        feature_names = self.feature_names_
+        vocab = self.vocabulary_
+        
         for x in X:
             for f, v in six.iteritems(x):
                 if isinstance(v, six.string_types):
                     f = "%s%s%s" % (f, self.separator, v)
                 if f not in vocab:
+                    self._validate_feature_name(f, x)
+                    if len(feature_names) == self.n_features:
+                        raise ValueError("The size of vocab is "
+                                         "larger than n_features.")
                     feature_names.append(f)
                     vocab[f] = len(vocab)
-
+                    
+        
         if self.sort:
             feature_names.sort()
             vocab = dict((f, i) for i, f in enumerate(feature_names))
+        
+        if getattr(self, "n_features", None) is not None:
+            for k in range(len(feature_names), self.n_features):
+                vocab[str(k)] = None
```

Comment:
I don't see why we need this, or why it's helpful when we want to call `partial_fit` again with a different set of features.